src/combinatorics/additive/salem_spencer.lean

```diff
@@ -0,0 +1,431 @@
+  apply nat.le_of_lt_succ,
+  change roth_number_nat (n + 1) < m.succ,
+  apply add_roth_number_lt_of_forall_not_add_salem_spencer,
+  simp only [range_succ, powerset_len_succ_insert not_mem_range_self, mem_union, mem_image,
+    or_imp_distrib, forall_and_distrib, and_imp, coe_insert, forall_exists_index,
+    forall_apply_eq_imp_iff₂],
+  refine ⟨_, λ s hs hsN, _⟩,
+  { simp only [mem_powerset_len, and_imp],
+    rw ←not_lt at hn,
+    exact λ x hx₁ hx₂ h, hn (h.le_roth_number_nat _ (λ x hx, mem_range.1 (hx₁ hx)) hx₂) },
+  simp only [and_imp, exists_prop, mem_filter, exists_and_distrib_left] at h,
+  obtain ⟨a, hNa, ha, haN⟩ := h s hs (hsN.mono $ set.subset_insert _ _),
+  rw [mem_powerset_len] at hs,
+  replace hs := hs.1 haN,
+  rw hsN (set.mem_insert_of_mem _ haN) (set.mem_insert _ _) (set.mem_insert_of_mem _ ha) _ at hs,
+  exact not_mem_range_self hs,
+  { rw [tsub_add_cancel_of_le hNa] }
+end
+
+/-- In conjunction with `dec_trivial`, this allows quick computation of small Roth numbers. -/
+lemma roth_number_nat_succ_eq {m n : ℕ} (hn : roth_number_nat n = m)
+  (h : ∀ s ∈ (powerset_len m (range n)).filter (λ (s : finset ℕ), add_salem_spencer (s : set ℕ)),
+          ∃ z ∈ s, n ≤ z + z ∧ z + z - n ∈ s) :
+  roth_number_nat (n + 1) = m :=
+(roth_number_nat_succ_le hn.le h).antisymm $ hn.ge.trans $ roth_number_nat.mono n.le_succ
+
+@[simp] lemma roth_number_nat_zero : roth_number_nat 0 = 0 := rfl
+@[simp] lemma roth_number_nat_one : roth_number_nat 1 = 1 := rfl
+@[simp] lemma roth_number_nat_two : roth_number_nat 2 = 2 := rfl
+@[simp] lemma roth_number_nat_three : roth_number_nat 3 = 2 := rfl
+@[simp] lemma roth_number_nat_four : roth_number_nat 4 = 3 := rfl
+@[simp] lemma roth_number_nat_five : roth_number_nat 5 = 4 := rfl
+@[simp] lemma roth_number_nat_six : roth_number_nat 6 = 4 := rfl
+@[simp] lemma roth_number_nat_seven : roth_number_nat 7 = 4 := rfl
+@[simp] lemma roth_number_nat_eight : roth_number_nat 8 = 4 := rfl
+@[simp] lemma roth_number_nat_nine : roth_number_nat 9 = 5 := rfl
+@[simp] lemma roth_number_nat_ten : roth_number_nat 10 = 5 := dec_trivial
+@[simp] lemma roth_number_nat_eleven : roth_number_nat 11 = 6 :=
+begin
+  apply le_antisymm,
+  { simpa using roth_number_nat_add_le 3 8 },
+  apply add_salem_spencer.le_roth_number_nat {0, 1, 3, 7, 8, 10},
+  { dec_trivial },
+  { simp },
+  { simp }
+end
+
+@[simp] lemma roth_number_twelve : roth_number_nat 12 = 6 :=
+roth_number_nat_succ_eq roth_number_nat_eleven $ by dec_trivial
+
+@[simp] lemma roth_number_nat_thirteen : roth_number_nat 13 = 7 :=
+begin
+  apply le_antisymm,
+  { simpa using roth_number_nat_add_le 10 3 },
+  apply add_salem_spencer.le_roth_number_nat {0, 1, 3, 4, 9, 10, 12},
+  { dec_trivial },
+  { simp },
+  { simp }
+end
+
+@[simp] lemma roth_number_nat_fourteen : roth_number_nat 14 = 8 :=
+begin
+  apply le_antisymm,
+  { simpa using roth_number_nat_add_le 11 3 },
+  apply add_salem_spencer.le_roth_number_nat {0, 1, 3, 4, 9, 10, 12, 13}, -- unique example
```

Comment:
I believe that the fact this is unique should help give bounds on `roth_number_nat 15`; since any example showing `roth_number_nat 15 = 9` must be of this form with 14 added on (which obviously doesn't work) - this was the motivation behind `roth_number_nat_succ_le`. If I remember correctly though, using that doesn't work; but we could prove this is the unique example instead... This is not important though, and you can add it as a TODO if you prefer.

Reply:
Taking that to #10645.